gensim/summarization/pagerank_weighted.py

```diff
@@ -89,40 +95,45 @@ def build_adjacency_matrix(graph):
     col = []
     data = []
     nodes = graph.nodes()
+    nodes2id = {v: i for i, v in enumerate(nodes)}
     length = len(nodes)
 
     for i in range(length):
         current_node = nodes[i]
-        neighbors_sum = sum(graph.edge_weight((current_node, neighbor)) for neighbor in graph.neighbors(current_node))
-        for j in range(length):
-            edge_weight = float(graph.edge_weight((current_node, nodes[j])))
-            if i != j and edge_weight != 0.0:
+        neighbors = graph.neighbors(current_node)
+        neighbors_sum = sum(graph.edge_weight((current_node, neighbor)) for neighbor in neighbors)
+        for neighbor in neighbors:
+            edge_weight = float(graph.edge_weight((current_node, neighbor)))
+            if edge_weight != 0.0:
                 row.append(i)
-                col.append(j)
-                data.append(edge_weight / neighbors_sum)
+                col.append(nodes2id[neighbor])
+                data.append(coeff * edge_weight / neighbors_sum)
```

Comment:
what's a reason to move dumping coeff here?

Reply:
-1 pass by array on len(edges) (and actually by matrix on (len(graph), len(graph) due to transforming from sparse to dense)